src/orange/orangeinp/detail/LocalSurfaceInserter.cc

```diff
@@ -7,19 +7,38 @@
 //---------------------------------------------------------------------------//
 #include "LocalSurfaceInserter.hh"
 
+#include "SurfaceHashPoint.hh"
+
 namespace celeritas
 {
 namespace orangeinp
 {
 namespace detail
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+real_type calc_length_scale(Tolerance<> const& tol)
+{
+    CELER_EXPECT(tol);
+    return tol.abs / tol.rel;
+}
+
+//---------------------------------------------------------------------------//
+}  // namespace
+
 //---------------------------------------------------------------------------//
 /*!
- * Construct with defaults.
+ * Construct with tolerance and a pointer to the surfaces vector.
+ *
+ * This builds an acceleration grid using the length scale of the problem
+ * (deduced from the tolerance) and the relative tolerance.
  */
 LocalSurfaceInserter::LocalSurfaceInserter(VecSurface* v,
                                            Tolerance<> const& tol)
-    : surfaces_{v}, soft_surface_equal_{tol}
+    : surfaces_{v}
+    , soft_surface_equal_{tol}
+    , calc_hashes_{real_type{0.01} * calc_length_scale(tol), 2 * tol.rel}
```

Comment:
I will make this "magic" grid spacing factor into a static constexpr variable for now; we could make it user-configurable in the future (but it will only change construction-time performance so probably isn't worth that).